icu4c/source/common/serv.cpp

```diff
@@ -941,7 +941,7 @@ ICUService::acceptsListener(const EventListener& l) const
 void 
 ICUService::notifyListener(EventListener& l) const 
 {
-    ((ServiceListener&)l).serviceChanged(*this);
+    (dynamic_cast<ServiceListener&>(l)).serviceChanged(*this);
```

Comment:
I am surprised that we can't be sure that this is a ServiceListener...?

All of the code has been working fine, so it should be safe to use static_cast throughout library code, rather than dynamic_cast, except for a very small number of specific cases where we have already needed dynamic_cast for type checking.